go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -361,6 +361,12 @@ func (qre *QueryExecutor) checkAccess(authorized *tableacl.ACLResult, tableName
 			qre.tsv.Stats().TableaclPseudoDenied.Add(statsKey, 1)
 			return nil
 		}
+
+		// Skip ACL check for queries against the dummy dual table
+		if tableName == "dual" {
+			return nil
```

Comment:
this is the fix (the rest of the changes were just an attempt to add a unit test which I couldn't make to pass due the weird use of the `fakedb` and query planner adding `limit 10001`

Reply:
imho, the fact that the unit test was not failing with the `acl error` indicates that the fix is correct